rocAL/source/decoders/audio/audio_decoder_factory.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright (c) 2019 - 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+#include "audio_decoder_factory.h"
+#include "decoder_factory.h"
+#include <audio_decoder.h>
+#include <sndfile_decoder.h>
+#include "commons.h"
+
+std::shared_ptr<AudioDecoder> create_audio_decoder(DecoderConfig config) {
+    switch (config.type()) {
+        case DecoderType::SNDFILE:
+            return std::make_shared<SndFileDecoder>();
+        default:
+            THROW("Unsupported decoder type " + TOSTR(config.type()));
+    }
+}
+
```

Comment:
Multiple lines at EOF

Reply:
reformatting done to all newly introduced files